### ISSUE TITLE
Rework SCA Policy for Windows Server 2019

### ISSUE THREAD

**Issue by 72nomada** [wazuh/wazuh#16724]
|Wazuh version| Component | 
|---| --- |
| 4.10.0 | SCA | 

Rework SCA Policy for Microsoft Windows Server 2016 RTM
- [x] Solve typos.
- [x] Increase check coverage. 
- [x] Improve rules implementation.
- [x] Improve texts.
- [ ] Solve reported issues.

- Fix the check "Ensure 'Domain member: Maximum machine account password age' is set to '30 or fewer days, but not 0'. (Automated)"
- Fix the check "Ensure 'Accounts: Guest account status' is set to 'Disabled' (MS only). (Automated)"
- https://github.com/wazuh/wazuh/issues/16479

**Comment by hayesb-emmanuel** [comment]
I don't believe this is exclusive to the 2019 test, other older OS versions such as 2012/2016 are the same I believe.

**Comment by 72nomada** [comment]
Hi @hayesb-emmanuel, Yes, You are right, we will manage other SCAs in different issues.

**Comment by Johnng007** [comment]
**18/12/2023**
Generation of YAML file and Preparing Test Environment
https://github.com/wazuh/wazuh/commit/ea3bd1b6c38f1d7d70401e49df647076673acbcf

**Comment by Johnng007** [comment]
19/12/2023
https://github.com/wazuh/wazuh/commit/45973c7a89f453a346c73ae2c0ab3759d234e06f
Windows server 2022 SCA rework Section 1.1.1 - 2.3.8.1 -

**Comment by Johnng007** [comment]
**20/12/2023**
https://github.com/wazuh/wazuh/commit/c694c1dc0fc07771c2603aa68d0e9fbd01973ee2
Windows server 2019 SCA rework Section 2.3.8.1 - 2.3.9.2

**Comment by ooniagbi** [comment]
To be reviewed by @IsExec

**Comment by IsExec** [comment]
## Review Update - 16/01/2024

## Document Description, Policy and Language
 
 - <details><summary>Document Description and Policy (Automated)   
     🔴  </summary>
          - Line 11 and 16: Kindly verify the correct version and date
    </details>

 - <details><summary> Language   
     🔴  </summary>
          - Please include the Language disclaimer: "Please note that the rules provide accurate results for Windows Server 2019 Operating Systems with the System language set to English. The SCA policy will work with other languages but the results will be less accurate due to some of the rules that depend on the System language."

    </details>

**Comment by Johnng007** [comment]
Issue Fix Update - 17/01/2024
.

**Comment by Johnng007** [comment]
Fix for Issue https://github.com/wazuh/wazuh/issues/21298


  # (L1) Ensure 'Domain member: Maximum machine account password age' is set to '30 or fewer days, but not 0'. (Automated)
  - id: 15521
    title: "Ensure 'Domain member: Maximum machine account password age' is set to '30 or fewer days, but not 0'."
    description: "This policy setting determines the maximum allowable age for a computer account password. By default, domain members automatically change their domain passwords every 30 days. If you increase this interval significantly so that the computers no longer change their passwords, an attacker would have more time to undertake a brute force attack against one of the computer accounts. The recommended state for this setting is: 30 or fewer days, but not 0. Note: A value of 0 does not conform to the benchmark as it disables maximum password age. Note #2: Some problems can occur as a result of machine account password expiration, particularly if a machine is reverted to a previous point-in-time state, as is common with virtual machines. Depending on how far back the reversion is, the older machine account password stored on the machine may no longer be recognized by the domain controllers, and therefore the computer loses its domain trust. This can also disrupt non-persistent VDI implementations, and devices with write filters that disallow permanent changes to the OS volume. Some organizations may choose to exempt themselves from this recommendation and disable machine account password expiration for these situations."
    rationale: "In Active Directory-based domains, each computer has an account and password just like every user. By default, the domain members automatically change their domain password every 30 days. If you increase this interval significantly, or set it to 0 so that the computers no longer change their passwords, an attacker will have more time to undertake a brute force attack to guess the password of one or more computer accounts."
    impact: "None - this is the default behavior."
    remediation: "To establish the recommended configuration via GP, set the following UI path to 30 or fewer days, but not 0: Computer Configuration\\Policies\\Windows Settings\\Security Settings\\Local Policies\\Security Options\\Domain member: Maximum machine account password age."
    references:
      - "https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/domain-member-maximum-machine-account-password-age"
    compliance:
      - cis: ["2.3.6.5"]
      - cis_csc_v8: ["5.2"]
      - cmmc_v2.0: ["IA.L2-3.5.7"]
      - pci_dss_v4.0: ["2.2.2", "8.3.5", "8.3.6", "8.6.3"]
      - soc_2: ["CC6.1"]
    condition: all
    rules:
	
	WRONG RULE
      - 'c:net.exe accounts -> n:Maximum password age (days):\s+(\d+) compare <= 30'
      - 'c:net.exe accounts -> n:Maximum password age (days):\s+(\d+) compare > 0'
	  
	CORRECT RULE
      - 'r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters -> MaximumPasswordAge'
      - 'r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters -> MaximumPasswordAge -> 30'
      - 'not r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters -> MaximumPasswordAge -> 0'
	  
	  
	  
	  
  # (L1) Ensure 'Accounts: Guest account status' is set to 'Disabled'. (Automated)
  - id: 15509
    title: "Ensure 'Accounts: Guest account status' is set to 'Disabled'."
    description: "This policy setting determines whether the Guest account is enabled or disabled. The Guest account allows unauthenticated network users to gain access to the system. The recommended state for this setting is: Disabled. Note: This setting will have no impact when applied to the Domain Controllers organizational unit via group policy because Domain Controllers have no local account database. It can be configured at the domain level via group policy, similar to account lockout and password policy settings."
    rationale: "The default Guest account allows unauthenticated network users to log on as Guest with no password. These unauthorized users could access any resources that are accessible to the Guest account over the network. This capability means that any network shares with permissions that allow access to the Guest account, the Guests group, or the Everyone group will be accessible over the network, which could lead to the exposure or corruption of data."
    impact: "All network users will need to authenticate before they can access shared resources. If you disable the Guest account and the Network Access: Sharing and Security Model option is set to Guest Only, network logons, such as those performed by the Microsoft Network Server (SMB Service), will fail. This policy setting should have little impact on most organizations because it is the default setting in Microsoft Windows 2000, Windows XP, and Windows Server™ 2003."
    remediation: "To establish the recommended configuration via GP, set the following UI path to Disabled: Computer Configuration\\Policies\\Windows Settings\\Security Settings\\Local Policies\\Security Options\\Accounts: Guest account status."
    references:
      - "https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/accounts-guest-account-status"
    compliance:
      - cis: ["2.3.1.2"]
      - cis_csc_v8: ["4.7"]
      - cis_csc_v7: ["16.8"]
      - iso_27001-2013: ["A.9.2.1"]
      - pci_dss_v3.2.1: ["2.1", "2.1.1"]
      - pci_dss_v4.0: ["2.2.2", "2.3.1"]
      - soc_2: ["CC6.3"]
    condition: any
    rules:
      - 'c:net user guest -> r:Account active\s+No'
      - "c:net user guest -> r:The user name could not be found."

**Comment by Johnng007** [comment]
**09-02-2024**
 - Rework/Issue fix 1.1.2 - 2.3.4.2
   https://github.com/wazuh/wazuh/pull/21434/commits/7397a9563e93cfb0888332b90e689f12fceb915c

**Comment by IsExec** [comment]
## Review Update - 09/02/2024

Todo based on https://github.com/wazuh/wazuh/issues/17598

- 1.1.5 (L1) Ensure 'Password must meet complexity requirements' is set to 'Enabled'. (Automated) - Not Implemented
-  Ensure IDs are regularized to reflect the fix above

**Comment by Johnng007** [comment]
27/02/2024
2.3.11.4 - Does not have a reference section from the CIS document so will remain unchanged.
2.3.11.9 - Does not have a reference section from the CIS document so will remain unchanged.
2.3.11.10 - Does not have a reference section from the CIS document so will remain unchanged.
2.3.17.2 - Does not have a reference section from the CIS document so will remain unchanged.

2.3.11.9 - Fixed
2.3.11.10 - Fixed
2.3.17.2 - Fixed
18.6.21.1  - Fixed

**Comment by IsExec** [comment]
## 2.3.10.1 (L1) Ensure 'Network access: Allow anonymous SID/Name translation' is set to 'Disabled'. (Automated)

### Status across SCA for different Windows Operating System

S/N | Operating   System | Status | Last Merged   version | Any Change   Required?
-- | -- | -- | -- | --
1 | Windows 10 | Not Implemented | Ongoing | No
2 | Windows 11 | Not Implemented | 4.8.0 | No
  | Windows Server 2012   (Non R2) | Implemented with a   different rule | Ongoing | Yes
3 | Windows Server 2012   R2 | Implemented | Ongoing | Yes
4 | Windows Server 2016 | Implemented with a   different rule | 4.8.1 | Yes (because of   default value)
5 | Windows Server 2019 | Implemented | Ongoing | Yes
6 | Windows Server 2022 | Implemented | 4.8.1 | Yes

### Observation

- The rule is disabled by default in windows 10 and 11 (gpedit) without any registry key value as expected in the CIS document. Please see below:
![image](https://github.com/wazuh/wazuh/assets/121132106/fbb46d72-2302-4f3e-a979-b16d190a1639)
![2_3_10_1](https://github.com/wazuh/wazuh/assets/121132106/a4df9202-be4a-412c-9d4a-b525a46b168e)

- Windows servers (2012 r2, 2012 non r2, 2016, 2019, 2022) have the rule disabled by default too, however, the CIS document mentioned the expected registry key value. The rule is disabled by default in Windows Server 2022 but no registry key representation.
![2_3_10_1](https://github.com/wazuh/wazuh/assets/121132106/36492a0b-f003-4a46-ab07-48750cae6d03)

![image](https://github.com/wazuh/wazuh/assets/121132106/1a362a16-477d-4f9e-839c-c4258a80baa0)
### Current Rule in Windows Server 2012 Non R2 and Windows Server 2016
    condition: all
    rules:
      - 'r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Lsa'
      - 'r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Lsa -> TurnOffAnonymousBlock'
      - 'r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Lsa -> TurnOffAnonymousBlock -> 0'
### Suggested Rule for Windows Server based on the CIS document
Rationale: The behaviour in Windows Server 2022 suggests that the rule could be disabled by default yet no registry key representation.
    condition: any
    rules:
      - 'not r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Lsa'
      - 'not r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Lsa -> TurnOffAnonymousBlock'
      - 'r:HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Lsa -> TurnOffAnonymousBlock -> 0'

**Comment by ooniagbi** [comment]
Set _2.3.10.1 (L1) Ensure 'Network access: Allow anonymous SID/Name translation' is set to 'Disabled'. (Automated)_ to **'Not implemented'** because the rule performance is unstable.

**Comment by IsExec** [comment]
# Windows Server 2019 Rework SCA Testing Report
**Condition**: The new SCA was benchmarked against old Windows server 2019 SCA.  

## Agent running on old SCA (Windows server 2019 SCA)

###  Endpoint spec

<details>
<summary>CPU/RAM</summary>

````console
vagrant@WIN-6G581DDLN5Q C:\Users\vagrant>systeminfo

Host Name:                 WIN-6G581DDLN5Q
OS Name:                   Microsoft Windows Server 2019 Standard Evaluation
OS Version:                10.0.17763 N/A Build 17763
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Server
OS Build Type:             Multiprocessor Free
Registered Owner:
Registered Organization:   gusztavvargadr
Product ID:                00431-10000-00000-AA044
Original Install Date:     7/22/2019, 11:14:56 PM
System Boot Time:          7/23/2024, 7:27:24 AM
System Manufacturer:       innotek GmbH
System Model:              VirtualBox
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 142 Stepping 10 GenuineIntel ~2112 Mhz
BIOS Version:              innotek GmbH VirtualBox, 12/1/2006
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC) Coordinated Universal Time
Total Physical Memory:     4,024 MB
Available Physical Memory: 3,153 MB
Virtual Memory: Max Size:  5,432 MB
Virtual Memory: Available: 4,631 MB
Virtual Memory: In Use:    801 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              N/A
````

**Comment by IsExec** [comment]
# Observations

## Invalid checks

**Comment by ooniagbi** [comment]
Merged to 4.10.0
